### DISCUSSION TITLE
Can't get create_default_admin (in config.json) to create the admin user.

### DISCUSSION THREAD

**Discussion by maspotts** [drakkan/sftpgo#1482] [Q&A]
I have tried creating an admin user by supplying this config:

```
{
    "sftpd": {
        "bind_port": 2022
    },
    "ftpd": {
        "bind_port": 2121,
        "tls_config": {
            "enabled": true,
            "cert_file": "chatbots/certs/certificate.crt",
            "key_file": "chatbots/certs/private.key"
        }
    },
    "httpd": {
        "bind_port": 8080,
        "tls_config": {
            "enabled": true,
            "cert_file": "chatbots/certs/certificate.crt",
            "key_file": "chatbots/certs/private.key"
        }
    },
  "data_provider": {
    "create_default_admin": true
  }
}

```
and starting sftpgo with:

`SFTPGO_DEFAULT_ADMIN_USERNAME=admin SFTPGO_DEFAULT_ADMIN_PASSWORD=fooby /opt/homebrew/opt/sftpgo/bin/sftpgo serve --config-file config.json --log-file-path sftpg.log
`

but I don't see any error or success messages related to creating the admin user, and I can't login with admin/fooby in the web UI.  If I can get past this (creating admin) then I'll be in a better place :)

**Comment by maspotts** [comment]
Is it appropriate to raise this as an issue? (expected behaviour: create admin)  But I suspect it's probably user error, which is why I'm asking here first...

**Comment by drakkan** [comment]
> Is it appropriate to raise this as an issue? (expected behaviour: create admin) But I suspect it's probably user error, which is why I'm asking here first...

I think there is some issue on your side. Based on your other questions I think you want to use the memory provider but please don't ask for step by step support as for our policy. Thanks for understanding

**Comment by maspotts** [comment]
Thanks: apologies for violating the policy; what would be an acceptable way to seek help in this case?  ie. should I be lumping all my questions into a single thread?  or are there other avenues I should explore before posting questions here?  (Not being snarky: I genuinely want to know how to ask for help but haven't been able to find any policy text to refer to!)

**Reply by drakkan** [reply]
Here is the support policy

https://github.com/drakkan/sftpgo?tab=readme-ov-file#support-policy

we have limited time and energy, so let's help the people/companies who support the project and thus contribute to keeping SFTPGo alive.

If you can't or don't want to support the project, feel free to post your support requests as Discussions and hope that someone will provide you with help. Thanks for understanding

**Comment by maspotts** [comment]
Got it: thanks!  Actually I am trying to deploy sftpgo as part of a solution for a service I'm creating for my company, and if I can get sftpgo to work then I was already planning to sign up as a sponsor: but my initial attempts to deploy it have all met with failure so far (can't create an admin user via the command-line; can't connect via HTTPS (using cert/key that's working fine in other services), so for now I had assumed that I could ask questions via this discussions forum (I definitely can't persuade our CFO to sign up as a sponsor of a project that I don't yet know if/how I can use!): however, in your first reply you asked me not to request 'step by step' support, so I'm wondering if I should be combining all my questions into a single thread?  ie. what's the bar for posting requests for help to this forum (in the pre-sponsorship/experimentation phase)?

**Comment by maspotts** [comment]
For context: sftpgo is the only project I've found which can implement the full functionality I'm looking for, ie:

  1. receive uploaded PDF files from remote networked scanners via HTTP(S), FTP(S) and SFTP;
  2. implement my custom business logic (assign a filename by generating a UUID and appending the timestamp and SHA2 hash);
  3. store the file in a designated S3 bucket
  4. deploy service via Docker
  
In the absence of sftpgo it looks like I'll have to deploy multiple services and/or write my own services, in order to achieve this same functionality, so I'm very motivated to get sftpgo working, since it would be a one-stop shop for me!  However, unfortunately in my initial experimentation I haven't been able to get it working on my own, and it doesn't seem to be a widely adopted project yet, so I was hoping that this community/discussion forum would enable me to get past my blockers and get me to the point where sftpgo is demonstrably working (as a prototype): if I can get to that point then I will happily sign up as a sponsor, because I much prefer to use/support an open source project (or write my own service) vs. a pay for a commercial, closed-source solution.  But if the sftpgo community is not big/mature enough to help a neophyte like myself get up and running, without signing up as a sponsor first, then I'm in a catch-22 situation and will have to fall back on my own resources.  I expect that my problems are due to my not fully understanding how to configure and launch the server, so was hoping that the questions I've posted so far would be easily resolved with some friendly hints or example code, but I'm very happy to conform to policy in order to avoid asking for step-by-step help if I know the preferred help protocol.  Apologies if I'm coming across negatively: that's not my intention: I'm just trying to clarify my goal and understand how (if) I can request help in order to improve my chances of achieving it...

**Comment by maspotts** [comment]
I've created a jumbo Q&A help request, describing my problem and asking for an example config.json, command-line, etc. in the hope that someone can get me past all my blockers in one step; apologies for pestering!  If this is not feasible I'll give up and go quietly :)
